### ISSUE TITLE
Interpreting the predicted result

### ISSUE THREAD

**Issue by npxhuy** [micnngo/tauFisher#1]
Hello,

First of all, thank you for developing this package. I was a bit puzzled about how to interpret the results.

In the example tutorial (multiple dataset case), I don’t quite understand why you train the data and then use it to predict on another dataset that already has the collection time points.

From what I understand, for example, if we train on a dataset with 5 time points and then test on another dataset with missing time points, tauFisher can predict the time at which each sample was taken. Is that the case?

If so, then why do we still need the parameter test_time = pseudobulk_time? It feels like we are predicting the time and then still validating it with the true time. I am a bit confused—please clarify.

If I was complete wrong on all of this, I hope you can help me answering the question on my own case, I have a data with 4 time points, can I train it, and predict the circadian time of a data with only 2 time points?

Best,
Huy

**Comment by seljd** [comment]
Hello Huy,

Thank you for using tauFisher!
> In the example tutorial (multiple dataset case), I don’t quite understand why you train the data and then use it to predict on another dataset that already has the collection time points.

In the tutorial, we are benchmarking tauFisher to evaluate its performance. For this, we use datasets with known collection time points so we can compare tauFisher’s prediction to the true time labels.

> From what I understand, for example, if we train on a dataset with 5 time points and then test on another dataset with missing time points, tauFisher can predict the time at which each sample was taken. Is that the case?

Yes, this is correct.

> If so, then why do we still need the parameter test_time = pseudobulk_time? It feels like we are predicting the time and then still validating it with the true time. I am a bit confused—please clarify.

You are correct that in the example we’re validating the predictions. In the `test_tauFisher` function, `test_time` is only used after predictions are made so you can display the true and predicted values side by side, making it easier to evaluate the performance. If your test data doesn’t have time labels, you can just use any numbers for `test_time` as placeholders. You may also check out the [MultipleDatasets_ByFunctions vignette](https://github.com/micnngo/tauFisher/blob/main/vignettes/MultipleDatasets_ByFunctions.Rmd) for a step-by-step walk-through.

> I have a data with 4 time points, can I train it, and predict the circadian time of a data with only 2 time points?

For the first part of the question: yes, you can use a dataset with 4 time points as training data, but the performance depends on how those time points are spaced. It is fine if they cover the 24-hour cycle well (e.g. ZT0, ZT6, ZT12 and ZT18, not necessarily evenly spaced). We recommend repeating ZT0 as ZT24 for better curve fitting outcome by FDA if the training data does not cover the entire 24-hour period (e.g. ZT0, ZT6, ZT12 and ZT18). However, if the time points are clustered together (e.g., ZT1, ZT2, ZT3, and ZT4), then the training data doesn’t capture the system’s 24-hour behavior, and the resulting predictions are unlikely to be accurate. 

For the second part of the question: Yes, the test data can have only two time points. tauFisher processes and makes predictions one test sample at a time, so having one or multiple test samples does not really matter.

I hope this helps and happy coding!

Closing this issue for now, but feel free to follow up. 
Junyan

**Comment by npxhuy** [comment]
Hello Junyan,

Thank you so much for your fast and detailed response.

And I have some follow up concerns that I hope you can help me clarify.

So we know that the sample collection time does not reflect the true time of the sample (circadian time) so that is why we want to predict it. And even though I wrote 

> tauFisher can predict the time at which each sample was taken

what I mean is, can tauFisher predict the inner time/circadian time the sample?

I am asking this because back to my own case, the 4 time points data (collected at 8, 12, 16, and 20) and trained it and use it to predict the circadian time of another experiment which only have two time points ( collected at 8 and 20). And after running tauFisher and the accuracy was 50%,  all the samples that were collected at 20 were predicted to have the circadian time at 8 or 9. Which it could be true although collecting at 20, the inner time of the sample was still at 8 or 9.
Another possible explanation is my train data might not give high accuracy, but that would be another story. 

Thanks again for your response!

Cheers,
Huy

**Comment by seljd** [comment]
Hello Huy,

> can tauFisher predict the inner time/circadian time the sample?

Yes, tauFisher predicts the inner time/circadian time of the sample. Since we are evaluating tauFisher's performance in the vignettes, we use datasets collected from control conditions so that the external time labels best reflect the inner time. The external time labels may not reflect the inner time if the system is disturbed.

> And after running tauFisher and the accuracy was 50%, all the samples that were collected at 20 were predicted to have the circadian time at 8 or 9.

This can happen when the rhythmic system in the testing data is different from the one in the training data. You can see two examples of tauFisher’s performance in disturbed systems in Supplementary Figure 2 of the [manuscript](https://doi.org/10.1038/s41467-024-48041-6). In the examples, tauFisher trained on the control system outputs similar values for the test samples from the disturbed systems, even though the test data were collected at different time points. Interestingly, tauFisher shows high accuracy when trained and tested on the same disturbed system.

Hope this helps and feel free to follow up!

Happy coding!
Junyan

**Comment by npxhuy** [comment]
Hello Junyan, 

Thank you for your respone.
I checked the Supplementary Figure 2.

> tauFisher trained on data collected from a disturbed system can predict time labels of samples collected from the same system.

So base on what I understand, when trained and tested on the same disturbed system, the predicted time was the collection time point and it was not the circadian time?

The second question is, the feeding time ZT5-ZT9 data was train and predicted on itself (trained on the first 2 replicates and predict on the third, for example), or ZT5-ZT9 was trained and predict on ZT0-ZT4 data?

> This can happen when the rhythmic system in the testing data is different from the one in the training data.

What could be considered as different in this case. Two mice groups that have different genotype, or different diet would also be considered as different?

Another follow up question is: do you think tauFisher can be use to apply on taxanomic abundance data, like instead of different gene it would be different taxa, and the measurement is the abundance of the taxa?

Cheers,
Huy

**Comment by seljd** [comment]
Hello Huy,
> So base on what I understand, when trained and tested on the same disturbed system, the predicted time was the collection time point and it was not the circadian time?

In the case in Supplementary Figure 2b, both the training and test data come from the same (but disturbed) system. tauFisher works by mapping the test data onto the training data, so the “predicted time” should be understood relative to the training set. We discuss this further in the Discussion section of the paper.

> The second question is, the feeding time ZT5-ZT9 data was train and predicted on itself (trained on the first 2 replicates and predict on the third, for example), or ZT5-ZT9 was trained and predict on ZT0-ZT4 data?

In Supplementary Figure 2a, the training data are from mice feeding in the control condition, and the test data are from mice feeding at ZT5-9 (left panel) and feeding at ZT0-4 (right panel). In Supplementary Figure 2b, the training and test data are kept within each feeding schedule: mice feeding at ZT5-9 (left panel), and mice feeding at ZT0-4 (right panel). Within each feeding schedule, we did leave-one-out (sample by sample) validation. More details are available in the main text and the figure legend.

> What could be considered as different in this case. Two mice groups that have different genotype, or different diet would also be considered as different?

Feeding influences the circadian clock, so in this case the time-restricted feeding schedules set at mice's resting phase disrupt their circadian clock. The only difference between the two groups of mice is their feeding schedule. More details on how those feeding schedules disrupt the clock in the skin are in this [paper](https://doi.org/10.1016/j.celrep.2017.07.022). 

> do you think tauFisher can be use to apply on taxanomic abundance data, like instead of different gene it would be different taxa, and the measurement is the abundance of the taxa?

It is an interesting question! I’m not very familiar with taxonomic abundance data, but in principle, if you have a list of robust circadian features (whatever the type) with their measurements, tauFisher should be able to make a prediction. It’s definitely worth trying!

Best,
Junyan
